src/transformers/models/auto/configuration_auto.py

```diff
@@ -1064,3 +1069,24 @@ def register(model_type, config, exist_ok=False):
                 "match!"
             )
         CONFIG_MAPPING.register(model_type, config, exist_ok=exist_ok)
+
+
+def sanitize_code_revision(pretrained_model_name_or_path, revision, trust_remote_code):
+    if revision in ["main", None] and not trust_remote_code:
+        revision_dict = {
+            "tiiuae/falcon-7b": "4e2d06f0a7c6370ebabbc30c6f59377ae8f73d76",
+            "tiiuae/falcon-7b-instruct": "f8dac3fff96d5debd43edf56fb4e1abcfffbef28",
+            "tiiuae/falcon-40b": "f1ba7d328c06aa6fbb4a8afd3c756f46d7e6b232",
+            "tiiuae/falcon-40b-instruct": "7475ff8cfc36ed9a962b658ae3c33391566a85a5",
+        }
+
+        if isinstance(pretrained_model_name_or_path, str) and pretrained_model_name_or_path.lower() in revision_dict:
+            revision = revision_dict.get(pretrained_model_name_or_path.lower())
+            logger.warning(
+                "The Falcon model was initialized without trust_remote_code=True, and will therefore leverage the "
+                "transformers library implementation. The repository revision is set to a version that doesn't "
+                "leverage remote code.\n\nIn order to override this, please set a revision manually or set "
+                "trust_remote_code=True."
+            )
```

Comment:
```suggestion
            logger.warning(
                "The Falcon model was initialized without `trust_remote_code=True`, and will therefore leverage the "
                f"transformers library implementation. {pretrained_model_name_or_path}'s revision is set to a version that doesn't "
                f"leverage remote code ({revision}).\n\nIn order to override this, please set a revision manually or set "
                "`trust_remote_code=True`."
            )
```
would just make it a little bit more informative if its not too much work

Reply:
I think the first line of this is a good idea, but 99% of the time this is triggered it'll be because the user didn't specify a `revision` kwarg, which means that the revision string will just print as `(None)`. Users might find that a bit confusing!